src/main/scala/mesosphere/marathon/core/instance/update/InstanceUpdateOpResolver.scala

```diff
@@ -39,7 +39,7 @@ private[marathon] class InstanceUpdateOpResolver(clock: Clock) extends StrictLog
           InstanceUpdateEffect.Update(updatedInstance, oldState = Some(oldInstance), Seq.empty)
         }
 
-      case RescheduleReserved(instance, runSpec) =>
+      case RescheduleReserved(_, runSpec) =>
```

Comment:
Ha, nice, I stumbled upon this as well. I actually think it should follow the same pattern as `schedule`. See our scheduling vs rescheduling logic in the [`LaunchQueueActor`](https://github.com/mesosphere/marathon/blob/master/src/main/scala/mesosphere/marathon/core/launchqueue/impl/LaunchQueueActor.scala#L269). What do you think?

Reply:
Schedule is CREATE this is UPDATE. No update operation can contain an instance object right now because it is not guaranteed that the object did not change in between because multiple updates might be queued. So I think the change I did there is in the right direction...